src/Repository/ProductCommentCriterionRepository.php

```diff
@@ -88,4 +253,116 @@ public function getByProduct($idProduct, $idLang)
 
         return $qb->execute()->fetchAll();
     }
+
+    /**
+     * Get Criterions
+     *
+     * @return array Criterions
+     */
+    public function getCriterions($id_lang, $type = false, $active = false)
+    {
+        $sql = '
+			SELECT pcc.`id_product_comment_criterion`, pcc.id_product_comment_criterion_type, pccl.`name`, pcc.active
+			FROM `' . _DB_PREFIX_ . 'product_comment_criterion` pcc
+			JOIN `' . _DB_PREFIX_ . 'product_comment_criterion_lang` pccl ON (pcc.id_product_comment_criterion = pccl.id_product_comment_criterion)
+			WHERE pccl.`id_lang` = ' . $id_lang . ($active ? ' AND active = 1' : '') . ($type ? ' AND id_product_comment_criterion_type = ' . (int) $type : '') . '
+			ORDER BY pccl.`name` ASC';
+        $criterions = $this->connection->executeQuery($sql)->fetchAll();
+
+        $types = self::getTypes();
+        foreach ($criterions as $key => $data) {
+            $criterions[$key]['type_name'] = $types[$data['id_product_comment_criterion_type']];
+        }
+
+        return $criterions;
+    }
+
+    /**
+     * @param int $id_criterion
+     *
+     * @return array
+     */
+    public function getProducts(int $id_criterion)
+    {
+        $sql = '
+			SELECT pccp.id_product, pccp.id_product_comment_criterion
+			FROM `' . _DB_PREFIX_ . 'product_comment_criterion_product` pccp
+			WHERE pccp.id_product_comment_criterion = ' . $id_criterion;
+
+        $res = $this->connection->executeQuery($sql)->fetchAll();
+
+        $products = [];
+        if ($res) {
+            foreach ($res as $row) {
+                $products[] = (int) $row['id_product'];
+            }
+        }
+
+        return $products;
+    }
+
+    /**
+     * @param int $id_criterion
+     *
+     * @return array
+     */
+    public function getCategories(int $id_criterion)
+    {
+        $sql = '
+			SELECT pccc.id_category, pccc.id_product_comment_criterion
```

Comment:
```suggestion
			SELECT pccc.id_category, pccc.id_product_comment_criterion
```
there's something wrong with indentation, I'm surprised cs fixer didn't catch it, you should indent using spaces, also the indentation itself is wrong for the queries

Reply:
Indentation and tabs are copied from [existing ObjectModel code](https://github.com/PrestaShop/productcomments/blob/v5.0.3/ProductCommentCriterion.php#L245-L248). I have just replaced tabs by spaces. Be free to change queries' indentation if you think it is needed.

Reply:
We know php-cs-fixer and php-stan are not reliable in some cases.